### PR TITLE
Add unit tests and fix empty config and mismatched braces bugs

### PR DESCRIPTION
**Pull request by GraceTsang**

### CODE REVIEW

File: config_parser.cc [hunk 1]
```diff
@@ -152,6 +152,7 @@ bool NginxConfigParser::Parse(std::istream* config_file, NginxConfig* config) {
   config_stack.push(config);
   TokenType last_token_type = TOKEN_TYPE_START;
   TokenType token_type;
+  int block_depth = 0;
```
Comment (avikbmohan) [inline config_parser.cc:155]: Maybe add a brief comment explaining what block_depth measures, and when it changes.

File: config_parser.cc [hunk 2]
```diff
@@ -199,18 +201,21 @@ bool NginxConfigParser::Parse(std::istream* config_file, NginxConfig* config) {
...
-          last_token_type != TOKEN_TYPE_END_BLOCK) {
+          last_token_type != TOKEN_TYPE_END_BLOCK &&
+	  last_token_type != TOKEN_TYPE_START) {
         // Error.
         break;
       }
-      return true;
+      if (block_depth == 0)
+          return true;
     } else {
```
Comment (avikbmohan) [inline config_parser.cc:221]: The varied indentations in the lines 212-218 above make the code slightly harder to read; I recommend standardizing them.

**Review by avikbmohan — COMMENTED** [review]
Also, Richard told me we can add a .gitignore to prevent the binaries from being included to keep things clean

File: config_parser.cc (hunk 1 above)
Comment (GraceTsang) [reply config_parser.cc:155]: Okay, added
